File: 1975. Maximum Matrix Sum/1975. Maximum Matrix Sum.js
/**
 * @param {number[][]} matrix
 * @return {number}
 */
var maxMatrixSum = function(matrix) {
    let totalSum = 0
    let min_abs = Infinity
    let negative_nums = 0
    for (row of matrix){
        for (val of row){
            totalSum += Math.abs(val)
            min_abs=Math.min(min_abs,Math.abs(val))
            if (val<0){
                negative_nums++
            }
        }
    }
    if (negative_nums%2===1){
        totalSum-=2*min_abs
    }
    return totalSum
};

console.log(maxMatrixSum(matrix = [[1,2,3],[-1,-2,-3],[1,2,3]]))

File: 1975. Maximum Matrix Sum/1975. Maximum Matrix Sum.py
from typing import List
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total_sum = 0
        min_abs_value = float('inf')  
        negative_count = 0
        for row in matrix:
            for value in row:
                total_sum += abs(value)  
                min_abs_value = min(min_abs_value, abs(value))  
                if value < 0:
                    negative_count += 1  
        
        if negative_count % 2 == 1:
            total_sum -= 2 * min_abs_value
        
        return total_sum

sol = Solution()
print(sol.maxMatrixSum(matrix = [[1,2,3],[-1,-2,-3],[1,2,3]]))